src/components/MarkerIconUpload.js

```diff
@@ -0,0 +1,83 @@
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { MediaUpload, MediaUploadCheck } from '@wordpress/block-editor';
+
+export default function MarkerIconUpload(props) {
+	const { marker, update } = props;
+	const ALLOWED_MEDIA_TYPES = ['image'];
+	const onUpdateImage = (media) => {
+		update({
+			...marker,
+			glyphImageId: media.id,
+			glyphImage: media.url,
+		});
+	};
+	const onRemoveImage = () => {
+		update({
+			...marker,
+			glyphImageId: undefined,
+			glyphImage: undefined,
+		});
+	};
+
+	return (
+		<div className="maps-block-apple-marker-uploader">
+			<MediaUploadCheck>
+				<label htmlFor="marker-icon-control">
```

Comment:
In order for this label to get all the propper styles from Gutenberg applied to it I would suggest using the [`BaseControl`](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/base-control) here as a wrapper :)

Reply:
Done. 👍